include/seqan3/argument_parser/detail/parse_format.hpp

```diff
@@ -0,0 +1,587 @@
+ *
+ * When parsing flags and options, the identifiers (and values) are removed from
+ * the vector parse_format::_argv. That way, options that are specified multiple times,
+ * but are no container type, can be identified and an error is reported.
+ */
+class parse_format
+{
+public:
+    /*!\brief The constructor
+     *
+     * \param argc The number of command line arguments
+     * \param argv The command line arguments
+     */
+    parse_format(int argc, char ** argv) :
+        _argc(argc)
+    {
+        init(argc, argv);
+    }
+
+    /*!\name Constructors
+     * \brief The (default) constructors 
+     * \{
+     */
+    parse_format() = delete;
+    parse_format(parse_format const & pf) = default;
+    parse_format & operator=(parse_format const & pf) = default;
+    parse_format(parse_format &&) = default;
+    parse_format & operator=(parse_format &&) = default;
+    //!\}
+
+    /*!\brief Adds an get_option call to be evaluated later on.
+     *
+     * \param value The variable in which to store the given command line argument.
+     * \param short_id The short identifier for the option (e.g. `'i'`).
+     * \param long_id The long identifier for the option (e.g. `"integer"`).
+     * \param desc The description of the option (which is only needed for the interface here).
+     * \param spec Advanced option specification. see seqan3::option_spec.
+     * \param validator The validator applied to the value after parsing (callable).
+     *
+     * \throws seqan3::parser_design_error
+     */
+    template <typename option_t, typename validator_type>
+    void add_option(option_t & value,
+                    char const short_id,
+                    std::string const & long_id,
+                    std::string const & desc,
+                    option_spec const & spec,
+                    validator_type && validator)
+    {
+        option_and_flag_calls.push_back([=, &value]()
+        {
+            get_option(value, short_id, long_id, spec, validator);
+        });
+    }
+
+    /*!\brief Adds a get_flag call to be evaluated later on.
+     *
+     * \param value The variable in which to store the given command line argument.
+     * \param short_id The short identifier for the flag (e.g. `'i'`).
+     * \param long_id The long identifier for the flag (e.g. `"integer"`).
+     * \param desc The description of the flag (which is only needed for the interface here).
+     * \param spec Advanced flag specification. see seqan3::option_spec.
+     *
+     * \throws seqan3::parser_design_error
+     */
+    void add_flag(bool & value,
+                  char const short_id,
+                  std::string const & long_id,
+                  std::string const & desc,
+                  option_spec const & spec)
+    {
+        option_and_flag_calls.push_back([=, &value]()
+        {
+            get_flag(value, short_id, long_id/*, spec*/); //TODO:: are specs needed? since a required flag makes no sense
```

Comment:
why should you ever require a flag? you can just set the bool to true yourself instead of asking the user for the flag